### PR TITLE
Use custom disconnect event instead of socket.io disconnect

### PR DESCRIPTION
**Pull request by hjheath**
Note this will only work once the server change has been released

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by hjheath** [comment]
This is now live on s3 staging so you can test it there @rfreitas

**Comment by rfreitas** [comment]
was not able to test the connection switch (due to staging constraints), however I was able to test disconnect and connecting again and in fact I didn't get the error screen.

However, when I closed the desktop (by closing the tab) I did not get the error screen, I believe this goes against expectations.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://240-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/index.js [hunk 1]
```diff
@@ -28,20 +28,23 @@ class CrossDeviceMobileRouter extends Component {
       token: null,
       steps: null,
       step: null,
-      socket: io(process.env.DESKTOP_SYNC_URL),
+      socket: io(process.env.DESKTOP_SYNC_URL, {autoConnect: false}),
```
Comment (rfreitas) [inline src/components/Router/index.js]: so if we don't have autoConnect how does the connection get re established?

Comment (hjheath) [reply src/components/Router/index.js]: All `autoConnect` does is decouple creating a socket and opening it. Reconnection logic is handled elsewhere.

From the [docs](https://socket.io/docs/client-api/#manager):
`autoConnect` (Boolean) by setting this `false`, you have to call manager.open whenever you decide it's appropriate
`reconnection` (Boolean) whether to reconnect automatically (`true`)

**Comment by seewah** [comment]
error spotted during testing
- turn on wifi
- open mobile link
- turn off wifi
- turn on 4g
- wait a few seconds
- see generic error screen

expected behaviour:
- no error screen
- user should be able to carry on with the flow

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://240-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
ok the issue reported above looks to have been fixed. thanks. I am accepting this ticket from my side

**Pull request merged**